packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1334,7 +1401,8 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
       // If we always have our own layer, then we can just repaint
       // ourselves without involving any other nodes.
       assert(_layer != null);
-      _nodesNeedingPaint.add(this);
+      if (owner != null)
+        owner._nodesNeedingPaint.add(this);
       Scheduler.instance.ensureVisualUpdate();
```

Comment:
We can wait for a later patch, but these calls to ensureVisualUpdate should be moved into the bindings layer because they're only needed if the owner is actually going to be visually displayed.